src/library/scala/collection/IndexedSeq.scala

```diff
@@ -25,6 +25,28 @@ trait IndexedSeq[+A] extends Seq[A]
   override protected[this] def stringPrefix: String = "IndexedSeq"
 
   override def iterableFactory: SeqFactory[IndexedSeq] = IndexedSeq
+
+  override def foldLeft[B](z: B)(f: (B, A) => B): B = {
+    var b = z
+    var i = 0
+    while (i < length) {
+      val a = apply(i)
+      i += 1
+      b = f(b, a)
```

Comment:
Small comment: in the codebase it is more customary to write the `i += 1` increment as the last opertion in the body of the loop.

Reply:
Noted. Will apply this in follow up PR.